src/generation/parser.rs

```diff
@@ -3465,13 +3465,41 @@ fn gen_reg_exp_literal(node: &Regex, _: &mut Context) -> PrintItems {
 }
 
 fn gen_string_literal<'a>(node: &'a Str, context: &mut Context<'a>) -> PrintItems {
+  let is_property_name = match node.parent() {
+    Node::KeyValueProp(parent) => match parent.key {
+      PropName::Str(_str) => true,
+      _ => false
+    }
+    Node::GetterProp(parent) => match parent.key {
+      PropName::Str(_str) => true,
+      _ => false
+    }
+    Node::SetterProp(parent) => match parent.key {
+      PropName::Str(_str) => true,
+      _ => false
+    }
+    Node::SetterProp(parent) => match parent.key {
+      PropName::Str(_str) => true,
+      _ => false
+    }
+    Node::MethodProp(parent) => match parent.key {
+      PropName::Str(_str) => true,
+      _ => false
+    }
+    _ => false
+  };
```

Comment:
We could save a few cycles by skipping this logic when `context.config.quote_props` is `Preserve` if you want.